### ISSUE TITLE
Is `name` actually optional?

### ISSUE THREAD

**Issue by Ethan-Arrowood** [openjs-foundation/package-json-research#9]
As mentioned [here](https://github.com/openjs-foundation/package-json-research/pull/3#discussion_r1484676870), there is reason to believe that the `name` property is not actually optional in practice. 

More research should be done regarding `npm` cli's usage of the property in various commands. If the lack of the `name` property causes anything but publishing to fail we must document that behavior.

**Comment by ljharb** [comment]
In particular, I suspect workspaces, `npm link`, `npm ls`, `npm explain`, are worth looking into.

**Comment by rotu** [comment]
FYI, I have seen `version` omitted when a `package.json` exists *only* to influence node behavior via `"type": "module" / "commonjs"`.

Node's docs seem to suggest this usage is kosher:
https://nodejs.org/api/packages.html#writing-dual-packages-while-avoiding-or-minimizing-hazards

If you have a `dist/esm` and `dist/cjs` folder and you want to keep the `.js` suffix, there doesn't seem to be a good way around it.

**Comment by ljharb** [comment]
At a project root, name and version are required; since all commands are meant to run in a project root, this works out fine. Indeed nested package.json files with only “type” or “main” are fine and expected.

**Comment by mcollina** [comment]
name is _mostly_ required, but I use `package.json` without it very often. It'd be odd to add names in those cases.

**Comment by wesleytodd** [comment]
I think for the sake of these docs we should elaborate on when it is required and when it is not. The goal is to educate and document, not to propose making it required when it is not actually required in all situations.

**Comment by Ethan-Arrowood** [comment]
Agreed - keep in mind that this documentation is not a specification. It may eventually lead to one someday, but for now we are solely focussed on documenting how things currently work across (the ever growing) JavaScript ecosystem.

**Comment by rotu** [comment]
Yeah, I think any sort of description of `package.json` has to be with respect to a tool. So the question “is name actually optional?” is underspecified.

For instance, `name` is required by `npm` but not required by `node`. Other tools (`pnpm`, `yarn`, `deno`, `bun`, `esbuild`, [SchemaStore](https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/package.json)…) have diverging expectations.

**Comment by wesleytodd** [comment]
Ideally we could make a compelling case for all of those projects to come together here to build our the right links or docs for folks doing cross ecosystem work.
